### PR TITLE
Support for cloudtrail logs

### LINKED ISSUE #102: SAR: support cloudtrail logs
**Describe the enhancement:**
Cloudtrail log entries are a single json object that we should extract the content of `Records` field from: it contains an array of json object that we should forward as events instead of the encompassing json log entry object

**Describe a specific use case for the enhancement or feature:**
Sending the json objects inside the array contained in `Records` field of the json log entry object of cloudtrail logs will allow the users to use the `aws.cloudtrail` integration

### PR DESCRIPTION
**Pull request by aspacca**
Enhancement
## What does this PR do?
Add support for extracting events from a json fields

## Why is it important?
Cloudtrail log entries are a single json object that we should extract the content of `Records` field from: it contains an array of json object that we should forward as events instead of the encompassing json log entry object

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~- [ ] I have made corresponding changes to the documentation~
~- [ ] I have made corresponding change to the default configuration files~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.md`

## Author's Checklist

- [ ]

## Related issues

-
Closes #102

### CODE REVIEW

File: handlers/aws/cloudwatch_logs_trigger.py [hunk 1]
```diff
@@ -94,25 +96,28 @@ def _handle_cloudwatch_logs_event(event: dict[str, Any], aws_region: str) -> Ite
...
             assert isinstance(log_event, bytes)
 
-            es_event = deepcopy(_default_event)
-            es_event["@timestamp"] = datetime.datetime.now().strftime("%Y-%m-%dT%H:%M:%S.%fZ")
-            es_event["fields"]["message"] = log_event.decode("UTF-8")
+            for extracted_log_event, extracted_starting_offset, is_last_event_extracted in extract_events_from_field(
+                log_event, json_object, starting_offset, ending_offset, integration_scope, extractor_events_from_field
+            ):
+                es_event = deepcopy(_default_event)
+                es_event["@timestamp"] = datetime.datetime.now().strftime("%Y-%m-%dT%H:%M:%S.%fZ")
```
Comment (zmoog) [inline handlers/aws/cloudwatch_logs_trigger.py:106]: I am not 100% sure about this, it's just a double check.

The use of the `Z` in the  `%Y-%m-%dT%H:%M:%S.%fZ` format implies this is a UTC timezone, right?

The use of `datetime.datetime.now()` to get a `datetime` object might return a local date and time.

Comment (aspacca) [reply handlers/aws/cloudwatch_logs_trigger.py:106]: good catch!

it is not only here, btw, but on every input handler

I've chosen for format mimicking filebeat, forgetting about the timezone implications

we should use `datetime.datetime.utcnow()`

Comment (zmoog) [reply handlers/aws/cloudwatch_logs_trigger.py]: I was bitten from Python's timezone support in the past ^^

**Review by zmoog — APPROVED** [review]

**Pull request merged**